### DISCUSSION TITLE
jinja2 error ModuleNotFoundError: No module named 'jinja2._compat'

### DISCUSSION THREAD

**Discussion by saulgoodmanverified** [pallets/jinja#1529] [Q&A] [answered]
Trying to upgrade jinja2 to the latest stable version

Error according to the logs: ModuleNotFoundError: No module named 'jinja2._compat'

Nothing related to this in the change logs https://jinja.palletsprojects.com/en/3.0.x/changes/#version-2-10

This is the sample code:
```
    from jinja2._compat import implements_to_string
    
    @implements_to_string
    def my_function():
        pass

```

**Comment by ThiefMaster** [comment, marked as answer]
That's **internal** code from the Python 2 era. It no longer exists in v3 which is Python-3-only.

**Reply by saulgoodmanverified** [reply]
So, just removing the decorator will work fine for Python-3?

**Reply by ThiefMaster** [reply]
I don't remember what it did, and it's used in your custom code. You need to check why you were using it and what the function was doing..
